homeassistant/components/weatherflow_cloud/sensor.py

```diff
@@ -176,35 +317,154 @@ async def async_setup_entry(
+            station_id
+        ]
+        for sensor_description in WEBSOCKET_WIND_SENSORS
+    )
 
-class WeatherFlowCloudSensor(WeatherFlowCloudEntity, SensorEntity):
-    """Implementation of a WeatherFlow sensor."""
+    entities.extend(
+        WeatherFlowWebsocketSensorObservation(
+            coordinator=observation_coordinator,
+            description=sensor_description,
+            station_id=station_id,
+            device_id=device_id,
+        )
+        for station_id in observation_coordinator.stations.station_outdoor_device_map
+        for device_id in observation_coordinator.stations.station_outdoor_device_map[
+            station_id
+        ]
+        for sensor_description in WEBSOCKET_OBSERVATION_SENSORS
+    )
+    async_add_entities(entities)
 
-    entity_description: WeatherFlowCloudSensorEntityDescription
+
+class WeatherFlowSensorBase(WeatherFlowCloudEntity, SensorEntity, ABC):
+    """Common base class."""
+
+    _attr_extra_state_attributes: dict[str, str]
 
     def __init__(
         self,
-        coordinator: WeatherFlowCloudDataUpdateCoordinator,
-        description: WeatherFlowCloudSensorEntityDescription,
+        coordinator: Any,
+        description: WeatherFlowCloudSensorEntityDescription
+        | WeatherFlowCloudSensorEntityDescriptionWebsocketWind
+        | WeatherFlowCloudSensorEntityDescriptionWebsocketObservation,
         station_id: int,
+        device_id: int | None = None,
     ) -> None:
-        """Initialize the sensor."""
-        # Initialize the Entity Class
+        """Initialize a sensor."""
         super().__init__(coordinator, station_id)
+        self.station_id = station_id
+        self.device_id = device_id
         self.entity_description = description
-        self._attr_unique_id = f"{station_id}_{description.key}"
+        self._attr_unique_id = self._generate_unique_id()
+
+    def _generate_unique_id(self) -> str:
+        """Generate a unique ID for the sensor."""
+        if self.device_id is not None:
+            return f"{self.station_id}_{self.device_id}_{self.entity_description.key}"
+        return f"{self.station_id}_{self.entity_description.key}"
+
+    @property
+    @abstractmethod
+    def native_value(self) -> StateType | date | datetime | Decimal:
+        """Abstract method for native value."""
+
+    @property
+    def available(self) -> bool:
+        """Get if available."""
+        if self.device_id is not None:
+            return bool(
+                self.coordinator.data
+                and self.coordinator.data[self.station_id][self.device_id] is not None
+            )
+        return bool(self.coordinator.data)
+
+
+class WeatherFlowWebsocketSensorObservation(WeatherFlowSensorBase):
+    """Class for Websocket Observations."""
+
+    entity_description: WeatherFlowCloudSensorEntityDescriptionWebsocketObservation
+    _attr_extra_state_attributes = {"Data source": WEBSOCKET_API}
```

Comment:
I don't think this is something that adds much for the user, I rather use the docs to explain this to users